src/shared/ui/Input/ui/index.tsx

```diff
@@ -24,14 +29,24 @@ const Input = (props: IInputCommonProps) => {
   }
 
   return (
-    <Form.Control
-      type={type}
-      name={name}
-      placeholder={placeholder}
-      value={value}
-      onChange={onInputChange}
-      className={classNames("mb-3", className)}
-    />
+    <div className={classNames("input-with-icon-wrap")}>
+      <Form.Control
+        type={type}
+        name={name}
+        placeholder={placeholder}
+        value={value}
+        onChange={onInputChange}
+        className={classNames("default-input", className, { "input-error-shadow": error })}
+      />
+
+      {Icon && onIconClick && (
```

Comment:
I would use here only Icon rendering and handle the click on it outside of the component.
Like: `{Icon ? Icon : null}`

Reply:
Done